transcendence-app/src/shared/db/postgres/utils.ts

```diff
@@ -36,28 +36,31 @@ export const makeQuery = async <T>(
   }
 };
 
-export function insertWithClient<T extends Record<string, any>>(
-  client: PoolClient,
-  table: table,
-  entity: T,
-) {
-  const { cols, params, values } = entityQueryMapper(entity);
-  const text = `INSERT INTO ${table}(${cols.join(', ')}) VALUES (${params.join(
-    ',',
-  )}) RETURNING *;`;
-  return client.query(text, values);
-}
-export function updateByIdWithClient<T extends Record<string, any>>(
-  client: PoolClient,
-  table: table,
-  id: string,
-  chatroom: T,
-) {
-  const { cols, values } = entityQueryMapper(chatroom);
-  const colsToUpdate = cols.map((col, i) => `${col}=$${i + 2}`).join(',');
-  const text = `UPDATE ${table} SET ${colsToUpdate} WHERE "id"=$1 RETURNING *;`;
-  return client.query(text, [id, ...values]);
-}
+export const makeTransactionalQuery = async <T>(
+  pool: PostgresPool,
+  callback: (client: PoolClient) => Promise<T | null>,
+) => {
+  let data: T | null = null;
```

Comment:
this variable could be one scope more restricted, i.e. it could be declared in the try block below.

Reply:
Updated. Thank you.